docs/programming_guide.md

```diff
@@ -2398,6 +2399,61 @@ of a function, or use the `withParameters(...)` method to pass in a configuratio
 [Back to top](#top)
 
 
+Passing parameters to functions
+-------------------
+
+Parameters can be passed to rich functions using either the constructor (if the function is
```

Comment:
Only the Configuration object method requires a RichFunction to overwrite `open()`. Configuring via the constructor (or any other method which sets non-transient fields) works also for functions that implement the function interfaces. Also it is not required that the class is defined as `static final`. That is only necessary if the class is defined inside another class.